pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/security/xml/ApexSharingViolations.xml

```diff
@@ -99,6 +99,63 @@ public inherited sharing class MyClass {
    public List<Contact> getAllTheSecrets() {
        return [SELECT Name FROM Contact];
    }
+}
+     ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Apex test classes do not need a sharing declaration</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+@isTest public class MyClass {
+   public List<Contact> getAllTheSecrets() {
+       return [SELECT Name FROM Contact];
+   }
+}
+     ]]></code>
+    </test-code>
+
+    <!-- @isTest can only be declared on the outer class, it is inherited by the inner class-->
+    <test-code>
+        <description>Apex inner test classes do not need a sharing declaration</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+@isTest public class Outer {
+   public List<Contact> getAllTheOuterSecrets() {
+       return [SELECT Name FROM Contact];
+   }
+   public class Inner {
+       public List<Contact> getAllTheInnerSecrets() {
+           return [SELECT Name FROM Contact];
+       }
+   }
+}
+     ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Nested method calls are detected</description>
+        <expected-problems>1</expected-problems>
+        <code><![CDATA[
+public class Dao {
+  static Map<String, List<SObject>> recordsMap = new Map<String, List<SObject>>();
+
+  public List<SObject> getRecords(String query) {
+    if (!recordsMap.containsKey(query)) {
+      recordsMap.put(query, Database.query(query));
+    }
+    return recordsMap.get(query);
+  }
+}
+     ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Trigger does not require sharing</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+trigger MyTrigger on Account(after insert, after update) {
+    [SELECT Id,(SELECT Id FROM Opportunities) FROM Account WHERE Id IN :Trigger.New];
 }
```

Comment:
I now, all the cases are in your permutation test class, but I'd still add the two examples from the original issue here as test cases. These rule tests are proof for me, that we fixed the original issue - and I don't see it yet....
Could you please add these two cases? Thanks!